modules/deployment/src/main/java/org/eclipse/xpanse/modules/deployment/DeployService.java

```diff
@@ -310,17 +321,31 @@ public void redeployService(DeployServiceEntity serviceToRedeploy) {
      * Perform rollback when deployment fails and destroy the created resources.
      */
     public void rollbackOnDeploymentFailure(DeployTask deployTask,
-                                            DeployServiceEntity deployServiceEntity) {
+            DeployServiceEntity deployServiceEntity,
+            ServiceConfigurationEntity serviceConfigurationEntity) {
         log.info("Performing rollback of already provisioned resources.");
         if (Objects.nonNull(deployServiceEntity.getDeployResourceList())
                 && !deployServiceEntity.getDeployResourceList().isEmpty()) {
             log.info("Rollback to destroy created resources for service with ID: {}",
                     deployTask.getId());
             deployTask.setDeploymentScenario(DeploymentScenario.ROLLBACK);
+            if (Objects.nonNull(serviceConfigurationEntity)) {
```

Comment:
This calling is duplicate with the calling `deleteServiceConfiguration(destroyTask);` in the method `destroy` (Line 502); because this method will call the method `destroy`.